### PR TITLE
Add C# 12 and .NET 8 polyfills

### PR DESCRIPTION
**Pull request by Sergio0694**
### Description

This PR adds the following polyfills for C# 12 and .NET 8:
- `[RequiresLocation]`
- `[CollectionBuilder]`
- `[UnsafeAccessor]`
- `[Experimental]`
- `[InlineArray]`

### CODE REVIEW

**Comment by dongle-the-gadget** [comment]
I thought `UnsafeAccessor` is not allowed within non .NET 8 binaries?

**Comment by Sergio0694** [comment]
Of course there is a plan, or this PR wouldn't exist. Just a little bit of patience, this is not my full time job 😅

File: README.md [hunk 1]
```diff
@@ -49,6 +49,9 @@ Here's an example of some of the new features that **PolySharp** can enable down
...
 - `[ModuleInitializer]` (needed to enable [custom module initializers](https://learn.microsoft.com/dotnet/csharp/language-reference/proposals/csharp-9.0/module-initializers))
+- `[RequiresLocation]` (needed to enable [ref readonly parameters](https://github.com/dotnet/csharplang/issues/6010))
+- `[CollectionBuilder]` (needed for [collection expressions](https://github.com/dotnet/csharplang/issues/5354))
+- `[Experimental]` (needed for [experimental features](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/proposals/csharp-12.0/experimental-attribute))
 
 To leverage them, make sure to bump your C# language version. You can do this by setting the `<LangVersion>` MSBuild property in your project. For instance, by adding `<LangVersion>11.0</LangVersion>` (or your desired C# version) to the first `<PropertyGroup>` of your .csproj file. For more info on this, [see here](https://sergiopedri.medium.com/enabling-and-using-c-9-features-on-older-and-unsupported-runtimes-ce384d8debb), but remember that you don't need to manually copy polyfills anymore: simply adding a reference to **PolySharp** will do this for you automatically.
```
Comment (WeihanLi) [inline README.md:56]: Maybe the `LangVersion` here could be updated to 12 also
```suggestion
To leverage them, make sure to bump your C# language version. You can do this by setting the `<LangVersion>` MSBuild property in your project. For instance, by adding `<LangVersion>12.0</LangVersion>` (or your desired C# version) to the first `<PropertyGroup>` of your .csproj file. For more info on this, [see here](https://sergiopedri.medium.com/enabling-and-using-c-9-features-on-older-and-unsupported-runtimes-ce384d8debb), but remember that you don't need to manually copy polyfills anymore: simply adding a reference to **PolySharp** will do this for you automatically.
```

**Comment by Trinitek** [comment]
> Note: waiting on [actions/runner-images#8797](https://github.com/actions/runner-images/issues/8797).

The image seems to be updated now.

**Comment by sliekens** [comment]
Hi @Sergio0694, is the namespace `System.Runtime.CompilerServices2` a typo or was it intended for the `InlineArrayAttribute`?

**Comment by sliekens** [comment]
Okay I see it's because `InlineArrayAttribute` visibility is always `public`, ignoring `PolySharpUsePublicAccessibilityForGeneratedTypes`. Is that a bug @Sergio0694 ?

**Comment by Sergio0694** [comment]
It is, yeah. I'll go fix it when I have a minute, good find! 😅

**Comment by sliekens** [comment]
If it's just a matter of changing the visibility and bumping the version, I can make a PR.

**Comment by Sergio0694** [comment]
Just needs changing the visibility, the version is set automatically by the CI when I create a release branch 🙂

**Comment by sliekens** [comment]
Voila #92

**Comment by AvremelM** [comment]
Sorry for the comment spam, but it's insane to me that C# 12 has been out for less than a month and you've already **released** polyfill support for it. 

Thank you so much @Sergio0694 (and collaborators!) for this project.

**Pull request merged**